cross/firebird/Makefile

```diff
@@ -41,11 +41,30 @@ INSTALL_TARGET = firebird_install
 # We must forward-declare the PKGVAR here as we must compile its value to the
 # binaries. We only support DSM7 so we do not worry about DSM6 compatibility
 SYNOPKG_PKGVAR = /var/packages/$(PKG_NAME)/var
+SYNOPKG_PKGDEST = /var/packages/$(PKG_NAME)/target
+
+ADDITIONAL_CFLAGS = '-DFB_PREFIX="\"$(SYNOPKG_PKGDEST)\""' \
+					'-DFB_SECDBDIR="\"$(SYNOPKG_PKGVAR)\""' \
+					'-DFB_GUARDDIR="\"$(SYNOPKG_PKGVAR)\""' \
+					'-DFB_CONFDIR="\"$(SYNOPKG_PKGVAR)\""' \
+					'-DFB_SBINDIR="\"$(SYNOPKG_PKGDEST)/bin\""' \
+					'-DFB_BINDIR="\"$(SYNOPKG_PKGDEST)/bin\""' \
+					'-DFB_DOCDIR="\"$(SYNOPKG_PKGDEST)/doc\""' \
+					'-DFB_HELPDIR="\"$(SYNOPKG_PKGDEST)/help\""' \
+					'-DFB_SAMPLEDIR="\"$(SYNOPKG_PKGDEST)/examples\""' \
+					'-DFB_PLUGDIR="\"$(SYNOPKG_PKGDEST)/plugins\""' \
+					'-DFB_SAMPLEDBDIR="\"$(SYNOPKG_PKGDEST)/examples/empbuild\""' \
+					'-DFB_TZDATADIR="\"$(SYNOPKG_PKGDEST)\""' \
+					'-DFB_MISCDIR="\"$(SYNOPKG_PKGDEST)/misc\""' \
+					'-DFB_MSGDIR="\"$(SYNOPKG_PKGDEST)\""' \
+					'-DFB_INCDIR="\"$(SYNOPKG_PKGDEST)/include\""' \
+					'-DFB_INTLDIR="\"$(SYNOPKG_PKGDEST)/intl\""' \
+					'-DFB_LIBDIR="\"$(SYNOPKG_PKGDEST)/lib\""' \
+					'-DFB_LOGDIR="\"$(SYNOPKG_PKGVAR)\""'
```

Comment:
If FirebirdSQL follows the known rules, you only have to define `FB_PREFIX` and other subfolders are defaulted to ${FB_PREFIX}/bin, ${FB_PREFIX}/lib, ...}

you should only require the variables
- `STAGING_INSTALL_PREFIX`
- `STAGING_INSTALL_PREFIX_VAR`

normally you do not define such FB_* variables as compiler flags but as configure options.

the configure arg `--prefix=$(INSTALL_PREFIX)` should make the definition of 'ADDITIONAL_CFLAGS += FB_PREFIX=$(INSTALL_PREFIX)` obsolete.


NB:
the spksrc framework is designed to use `/usr(local/{packagename}` as prefix when building a package with the cross Makefile and to use `/var/packages/{packagename}/target` as prefix when building the package in the spk folder.

Reply:
> for Makefiles you have to use the Makefile Variables as documented in the wiki
like `STAGING_INSTALL_PREFIX`

As I understand, the `STAGING_INSTALL_PREFIX` is the location where. the files should be copied to be found by the PLIST magic. What I need here is actually the runtime location of the files, because Firebird compiles this information in the binary. If not set properly, it does not find the configuration etc.

> If FirebirdSQL follows the known rules, you only have to define FB_PREFIX and other subfolders are defaulted to ${FB_PREFIX}/bin, ${FB_PREFIX}/lib, ...}

This works to a certain extent. With this setup, configuration files are located in `$FB_PREFIX`, where the installer does not find them. Even if I move them later to `$FB_PREFIX/var`, the Firebird itself won't find them during startup and crashes.

> normally you do not define such FB_* variables as compiler flags but as configure options.

This is true, but the Firebird's build process is non-standard at least. The server is built in two phases, first some native part is built, this is then used to pre-process certain files and then the actual cross toolchain is used to build the binary.

In this process, the `--with-fbprefix` and similar configure option have only limited usability, because the value from them does not reach the cross phase for the reason you mention in your next comment. The file that should be generated with `autoconf` is not generated, which is the destination location of the discussed variables.